src/tests/ftest/control/ms_resilience.py

```diff
@@ -0,0 +1,244 @@
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+from apricot import TestWithServers
+from general_utils import stop_processes
+from avocado.core.exceptions import TestFail
+import random
+import socket
+
+
+class ManagementServiceResilience(TestWithServers):
+    """Test Class Description:
+    Verify that MS is accessible as long as one instance is left.
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Inititialize a ManagementServiceResilience object."""
+        super(ManagementServiceResilience, self).__init__(*args, **kwargs)
+        self.setup_start_servers = False
+
+    def update_and_verify(self, ignore_status=False):
+        """Create a pool on the server group
+
+        Args:
+            ignore_status (bool): If False, raise an exception when pool create
+                fails, otherwise don't raise an exception. Defaults to False.
+        """
+        self.add_pool(create=False)
+        self.pool.name.value = self.server_group
+        try:
+            self.pool.create()
+        except TestFail as error:
+            if ignore_status:
+                self.log.info("Expected MS error creating pool!: %s", error)
+            else:
+                self.fail("Pool create failed unexpectedly!")
+
+        if not ignore_status:
+            self.log.info("Pool UUID %s on server group: %s",
+                          self.pool.uuid, self.server_group)
+            # Verify that the pool persisted.
+            pool_data = self.dmg.pool_list()
+            if pool_data and self.pool.uuid in pool_data:
+                self.log.info("Found pool in system.")
+            else:
+                self.fail("No pool found in system.")
+
+    def verify_leader(self, access_list):
+        """Verify the leader of the MS is in the access_list.
+
+        Args:
+            access_list (list): list of hostname representing the access points
+                for the MS.
+
+        Returns:
+            str: hostname of the MS leader. If the leader is not within the
+                access_list, the test will fail.
+
+        """
+        l_addr = self.dmg.system_leader_query()["leader"]
```

Comment:
(pylint-no-member) Instance of 'ManagementServiceResilience' has no 'dmg' member